src/coprocessor/dag/expr/mod.rs

```diff
@@ -281,6 +273,17 @@ impl Expression {
     }
 }
 
+pub fn decode<'a>(bytes: Cow<'a, [u8]>, chrst: &str) -> Result<Cow<'a, str>> {
```

Comment:
why we need this function? And the function name also makes me confused.

Reply:
IMHO it's better to separate `eval_string_and_decode` into `eval_string` and `decode`. I don't get why its name is confusing